salt/minion.py

```diff
@@ -198,6 +199,20 @@ def resolve_dns(opts, fallback=True):
     return ret
 
 
+def srv_lookup(name, protocol='TCP', domain=None):
+    if domain:
+        domain = '.' + domain
+    record_name = '_{0}_._{1}{2}'.format(name, protocol.lower(), domain or '')
```

Comment:
Format is: _service._proto.name.

Just need to remove the single '_' after {0}.